src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -328,12 +325,18 @@ internal open class ParallelThreadsRunner(
             // We do not want the transformed code to be reachable outside of the runner and strategy classes.
             return CompletedInvocationResult(collectExecutionResults(afterInitStateRepresentation, afterParallelStateRepresentation, afterPostStateRepresentation))
         } catch (e: TimeoutException) {
-            val threadDump = collectThreadDump(this)
-            return RunnerTimeoutInvocationResult(threadDump, collectExecutionResults())
+            return RunnerTimeoutInvocationResult()
         } catch (e: ExecutionException) {
+            // in case if invocation thrown an exception,
+            // make sure all threads terminate before continuing
+            // and handle possible hangings gracefully
+            runCatching { strategy.awaitAllThreads(timeout) }.onFailure { exception ->
+                if (exception is TimeoutException) return RunnerTimeoutInvocationResult()
```

Comment:
Nothing should happen. This is a case when scenario execution raised some exception. 
In this case, we just want to wait for all threads to terminate. 

However, the threads may hung. In this case, `awaitAllThreads` should throw `TimeoutException`.
So I wrap `awaitAllThreads` call into `runCatching` to handle this timeout exception and return `RunnerTimeoutInvocationResult`. 

Otherwise, if threads terminate normally without hunging, the `UnexpectedExceptionInvocationResult` will be returned with the original exception thrown by the scenario.

Reply:
Thanks for the clarification! Could you please add a short version of this comment to the code?